### ISSUE TITLE
0.18.4: DRM w/GLES: cog_drm_gles_renderer_handle_egl_image: Cannot create framebuffer  (Bad file descriptor)

### ISSUE THREAD

**Issue by deanmsands3** [Igalia/cog#748]
Distro: Debian Bullseye
OS: Linux 5.10.198
Arch: Arm64
GPU: Mali Midgard T-860
Driver: libmali-midgard-t86x-r18p0-wayland-gbm
Cog: 0.18.4
- Compiled from source
 - COG_ENABLE_WESTON_DIRECT_DISPLAY
 - COG_USE_SOUP2
 - Args: --platform=drm --platform-params='renderer=gles'

User: root

Hello!
Trying with the DRM platform this time.
Since only the ES2 variants of GLMark2 work, I'm specifying the GLES renderer.
It works for a few seconds then freezes.
The strace shows that the software can't open anymore files.
I've included an lsof log that confirms this.
It complains about not finding rules/evdev but I believe that's a symptom not a cause. The file is there.

How do I get this running?
Thanks in advance.

(The simple log is from a later run with the same outcome.)
Simple log: [root-cog-drm-gles-0.18.4.log](https://github.com/user-attachments/files/18096719/root-cog-drm-gles-0.18.4.log)
strace log: [root-strace-cog-drm-gles-0.18.4.log](https://github.com/user-attachments/files/18096659/root-strace-cog-drm-gles-0.18.4.log)
lsof log: [root-lsof-cog-drm-gles-0.18.4.txt](https://github.com/user-attachments/files/18096658/root-lsof-cog-drm-gles-0.18.4.txt)

**Comment by jrosa-digiwest** [comment]
I am facing the same issue on a **raspberry pi 4**. Do you have any updates? I have tested with cog `0.16.1` (from debian sources) and with cog `0.18.4` build from source.

Distro: Debian GNU/Linux 12 (bookworm)
OS: Linux 6.6.62+rpt-rpi-v8
Arch: aarch64
wpewebkit-driver: 2.38.6-1
libwpe-1.0-1: 1.14.0-1
libwpebackend-fdo-1.0-1: 1.14.2-1
cog: 0.16.1 and 0.18.4

**Comment by deanmsands3** [comment]
> I am facing the same issue on a **raspberry pi 4**. Do you have any updates? I have tested with cog `0.16.1` (from debian sources) and with cog `0.18.4` build from source.
> 
> Distro: Debian GNU/Linux 12 (bookworm) OS: Linux 6.6.62+rpt-rpi-v8 Arch: aarch64 wpewebkit-driver: 2.38.6-1 libwpe-1.0-1: 1.14.0-1 libwpebackend-fdo-1.0-1: 1.14.2-1 cog: 0.16.1 and 0.18.4

This fix won't work for me, but let me suggest you run it inside cage kiosk.
`cage -- cog --platform=wl "https://www.github.com"`
I also have an RPi4 and cog w/drm failed after a little while. But it's running ok with cage.
Unfortunately, our project calls for a TinkerBoard 2, and this doesn't work.

**Comment by jrosa-digiwest** [comment]
> > I am facing the same issue on a **raspberry pi 4**. Do you have any updates? I have tested with cog `0.16.1` (from debian sources) and with cog `0.18.4` build from source.
> > Distro: Debian GNU/Linux 12 (bookworm) OS: Linux 6.6.62+rpt-rpi-v8 Arch: aarch64 wpewebkit-driver: 2.38.6-1 libwpe-1.0-1: 1.14.0-1 libwpebackend-fdo-1.0-1: 1.14.2-1 cog: 0.16.1 and 0.18.4
> 
> This fix won't work for me, but let me suggest you run it inside cage kiosk. `cage -- cog --platform=wl "https://www.github.com"` I also have an RPi4 and cog w/drm failed after a little while. But it's running ok with cage. Unfortunately, our project calls for a TinkerBoard 2, and this doesn't work.

Yes, it worked with cage and wayland (wl). However, I can't manage to get it to work with DRM even when using cage. I wanted to use DRM to avoid overheads.
